### PR TITLE
K8SPSMDB-322: Add mongos status

### PR DESCRIPTION
**Pull request by blez**
[K8SPSMDB-322](https://jira.percona.com/browse/K8SPSMDB-322)

### CODE REVIEW

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-489-e5427afb

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| storage | passed |
| upgrade | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| limits | passed |
| upgrade-consistency | passed |
| arbiter | passed |
| scaling | passed |
| demand-backup | passed |
| operator-self-healing | passed |
| security-context | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| scheduled-backup | passed |
| version-service | passed |
| demand-backup-sharded | passed |
| users | passed |

File: pkg/apis/psmdb/v1/psmdb_types.go [hunk 1]
```diff
@@ -88,6 +88,13 @@ type ReplsetMemberStatus struct {
 	Version string `json:"version,omitempty"`
 }
 
+type MongosStatus struct {
+	Size    int      `json:"size"`
+	Ready   int      `json:"ready"`
+	Status  AppState `json:"status,omitempty"`
+	Message string   `json:"message,omitempty"`
+}
+
```
Comment (delgod) [inline pkg/apis/psmdb/v1/psmdb_types.go:97]: can we also modify CRD a bit, similar to PXC
https://github.com/percona/percona-xtradb-cluster-operator/blob/master/deploy/crd.yaml#L55-L58

**Review by delgod — CHANGES_REQUESTED** [review]

**Review by delgod — APPROVED** [review]

**Review by qJkee — APPROVED** [review]

File: pkg/apis/psmdb/v1/psmdb_types.go (hunk 1 above)
Comment (blez) [reply pkg/apis/psmdb/v1/psmdb_types.go:97]: Will be done in next release

**Review by nonemax — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| storage | passed |
| upgrade | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| limits | passed |
| upgrade-consistency | passed |
| scaling | passed |
| demand-backup | passed |
| arbiter | passed |
| security-context | passed |
| operator-self-healing | passed |
| service-per-pod | passed |
| scheduled-backup | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| demand-backup-sharded | failed |

**Pull request merged**